lib/gtfs/block.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Gtfs.Block do
+  alias Gtfs.Helpers
+  alias Gtfs.Trip
+
+  @type id :: String.t()
+  @type t :: [Trip.t()]
+  @type by_id :: %{{id(), Service.id()} => t()}
+
+  @spec group_trips_by_block([Trip.t()]) :: by_id()
+  def group_trips_by_block(trips) do
+    trips
+    |> Enum.filter(fn trip -> trip.stop_times != [] end)
+    |> Enum.group_by(fn trip -> {trip.block_id, trip.service_id} end)
+    |> Helpers.map_values(fn trips ->
+      Enum.sort_by(
+        trips,
+        fn trip -> List.first(trip.stop_times).time end
+      )
+    end)
+  end
+
+  @spec get(by_id(), id(), Service.id()) :: t() | nil
+  def get(by_id, block_id, service_id) do
```

Comment:
Rather than testing this function independently, I've added tests that check that `group_trips_by_block` and `get` work together when used in sequence so I don't have to make a bunch of maps that depend on implementation details.